### PR TITLE
Added Cassandra CQL binding.

### PR DESCRIPTION
**Pull request by cmatser**
Cassandra CQL is the newer query language for accessing Cassandra DB.  It's been around for a while and seems to be the language of choice going forward with Cassandra.  It's very similar to SQL.  As such, the learning curve to using Cassandra is easier through CQL.

### CODE REVIEW

**Comment by busbey** [comment]
thanks for the contribution and sorry for the lag. I intend to merge this in by the end of the week. I just have a couple of feedback points.

If you're not still working on this, I can make the modifications myself.

File: pom.xml [hunk 1]
```diff
@@ -67,7 +68,7 @@
     <module>dynamodb</module>
     <module>elasticsearch</module>
     <!--<module>gemfire</module>-->
-    <module>infinispan</module>
```
Comment (busbey) [inline pom.xml]: please don't include changes to other modules. if infinispan currently isn't building, we can disable it as a different PR.

File: pom.xml [hunk 2]
```diff
@@ -45,7 +45,8 @@
   <properties>
     <maven.assembly.version>2.2.1</maven.assembly.version>
     <hbase.version>0.92.1</hbase.version>
-    <cassandra.version>0.7.0</cassandra.version>
+    <cassandra.version>1.2.9</cassandra.version>
```
Comment (busbey) [inline pom.xml]: does this change mean we can't work with pre-1.2 servers?

File: pom.xml (hunk 1 above)
Comment (cmatser) [inline pom.xml]: Good point.  Please update this PR for me.

File: pom.xml (hunk 2 above)
Comment (cmatser) [inline pom.xml]: I added the new binding as a separate class CassandraCQLClient and left the previous bindings untouched.  So, you can use those with pre-1.2 servers.  But this property is used in the maven build to pull in the right dependencies for the new CassandraCQLClient.  I don't know if the dependencies for the pre-1.2 version are still being pulled in with this change.

I suppose I could have created a "cassandracql.version" property?  Then both old and new jars would be pulled, but I'm skeptical about that working.  Some dependency jars could get clobbered with older or newer versions.  I might tend towards creating it as a separate binding completely, with its own directory if we need old and new to work side-by-side.

I'm not sure this answers your question.  The folks at Datastax told me that Cassandra CQL was the direction of the data store moving forward.  They did not recommend using the older binding.  I did not test with it.

Thanks for your work!  Unfortunately, I don't have access to the environment I had when I did this work, but I'll help as much as I can.

Comment (busbey) [inline pom.xml]: that helps a bunch, thanks for getting back to me.

I suspect separate bindings are the way to go, but I agree that still working with older versions can be something we worry about later.

**Pull request closed without merging**